src/Agent.Listener/CommandSettings.cs

```diff
@@ -436,117 +453,230 @@ public string GetWork()
         }
 
         public string GetProxyUserName()
         {
-            return GetArg(Constants.Agent.CommandLine.Args.ProxyUserName);
+            return GetArg(Configure?.ProxyUserName, Constants.Agent.CommandLine.Args.ProxyUserName);
         }
 
         public string GetProxyPassword()
         {
-            return GetArg(Constants.Agent.CommandLine.Args.ProxyPassword);
+            return GetArg(Configure?.ProxyPassword, Constants.Agent.CommandLine.Args.ProxyPassword);
         }
 
         public bool GetSkipCertificateValidation()
         {
-            return TestFlag(Constants.Agent.CommandLine.Flags.SslSkipCertValidation);
+            return TestFlag(Configure?.SslSkipCertValidation , Constants.Agent.CommandLine.Flags.SslSkipCertValidation);
         }
 
         public string GetCACertificate()
         {
-            return GetArg(Constants.Agent.CommandLine.Args.SslCACert);
+            return GetArg(Configure?.SslCACert, Constants.Agent.CommandLine.Args.SslCACert);
         }
 
         public string GetClientCertificate()
         {
-            return GetArg(Constants.Agent.CommandLine.Args.SslClientCert);
+            return GetArg(Configure?.SslClientCert, Constants.Agent.CommandLine.Args.SslClientCert);
         }
 
         public string GetClientCertificatePrivateKey()
         {
-            return GetArg(Constants.Agent.CommandLine.Args.SslClientCertKey);
+            return GetArg(Configure?.SslClientCertKey, Constants.Agent.CommandLine.Args.SslClientCertKey);
         }
 
         public string GetClientCertificateArchrive()
         {
-            return GetArg(Constants.Agent.CommandLine.Args.SslClientCertArchive);
+            return GetArg(Configure?.SslClientCertArchive, Constants.Agent.CommandLine.Args.SslClientCertArchive);
         }
 
         public string GetClientCertificatePassword()
         {
-            return GetArg(Constants.Agent.CommandLine.Args.SslClientCertPassword);
+            return GetArg(Configure?.SslClientCertPassword, Constants.Agent.CommandLine.Args.SslClientCertPassword);
+        }
+
+        public bool GetGitUseSChannel()
+        {
+            return TestFlag(Configure?.GitUseSChannel, Constants.Agent.CommandLine.Flags.GitUseSChannel);
+        }
+
+        public bool GetEnvironmentVMResource()
+        {
+            return TestFlag(Configure?.EnvironmentVMResource, Constants.Agent.CommandLine.Flags.Environment);
+        }
+
+        public bool GetRunOnce()
+        {
+            return TestFlag(Run?.RunOnce, Constants.Agent.CommandLine.Flags.Once);
+        }
+
+        public bool GetDeploymentPool()
+        {
+            return TestFlag(Configure?.DeploymentPool, Constants.Agent.CommandLine.Flags.DeploymentPool);
+        }
+
+        public bool GetDeploymentOrMachineGroup()
+        {
+            if (TestFlag(Configure.DeploymentGroup, Constants.Agent.CommandLine.Flags.DeploymentGroup) || 
```

Comment:
you have a null check for Configure after using it directly. Seems like you need the null check to come earlier.